### PR TITLE
less info for api argument validation

### PR DESCRIPTION
**Pull request by au2**
## What was the problem?

MethodArgumentNotValidException, which is raised when there are rest end-point validation, included stack information. This caused concern in WASA scans.

Associated tickets or Slack threads:

- [MCP-1992](https://amida.atlassian.net/browse/MCP-1992)

## How does this fix it?[^1]

Handles MethodArgumentNotValidException specifically and only  field name and reason of error for the field is put into the error message. Also it now returns 400 (Bad Request) for these errors instead of 500.

## How to test this PR
- Open swagger and enter erroneous inputs. Check there is no stack information.
- Step 2[^secrel]

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: app/src/test/java/gov/va/vro/VroControllerTest.java [hunk 1]
```diff
@@ -124,34 +123,7 @@ void postFullHealthAssessment() throws Exception {
...
-                .interceptSendToEndpoint(
-                    "rabbitmq:health-assess-exchange"
-                        + "?routingKey=health-assess.7101&requestTimeout=60000")
-                .skipSendToOriginalEndpoint()
-                .to("mock:claim-submit-full"));
-
-    mockFullHealthEndpoint.whenAnyExchangeReceived(
-        FunctionProcessor.<Claim, String>fromFunction(claim -> util.claimToResponse(claim, false)));
-
+  void fullHealthAssessmentMissingEvidence() throws Exception {
```
Comment (wlin2019) [inline app/src/test/java/gov/va/vro/VroControllerTest.java:126]: It looks like this test failed.
Gradle Test Run :app:test > Gradle Test Executor 3 > VroControllerTest > fullHealthAssessmentMissingEvidence() FAILED
[150](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/3334337735/jobs/5517189235#step:5:151)
    org.opentest4j.AssertionFailedError at VroControllerTest.java:134

Comment (au2) [reply app/src/test/java/gov/va/vro/VroControllerTest.java:126]: @wlin2019 I cannot reproduce this locally and I do not think build check caught it either. Can you pull and try again?

**Comment by ianwolson** [comment]
It says it fails on line 138 in the VroControllerTest: 

> assertEquals(HttpStatus.NOT_FOUND, responseEntity.getStatusCode()); 

Maybe the status needs to be BAD_REQUEST like the other ones? Have not tried yet

File: app/src/test/java/gov/va/vro/VroControllerTest.java (hunk 1 above)
Comment (au2) [reply app/src/test/java/gov/va/vro/VroControllerTest.java:130]: @wlin2019 I put everything back. not sure why that fixes it but it does. so please take another look.

Comment (wlin2019) [reply app/src/test/java/gov/va/vro/VroControllerTest.java:130]: It works now.

**Review by wlin2019 — APPROVED** [review]
LGTM

**Review by ianwolson — APPROVED** [review]

**Pull request merged**